src/main/java/fr/insee/pearljam/api/controller/UserController.java

```
@@ -78,145 +76,111 @@ public ResponseEntity<UserDto> getUserById(
-		return new ResponseEntity<>(updatedUser, HttpStatus.OK);
+
+		UserDto updatedUser = userService.updateUser(user);
+		log.info("{} updated user {} - {} ", userId, userId, HttpStatus.OK.value());
+		return updatedUser;
 	}
 
 	/**
 	 * This method assign a user to target Organization Unit
-	 * 
+	 * @param userId user to assign to the organization
+	 * @param ouId organization unit id
+	 * @return the updated user
 	 */
 	@Operation(summary = "Assign User to Organization Unit")
 	@PutMapping(Constants.API_USER_ID_ORGANIZATION_ID_OUID)
-	public ResponseEntity<Object> assignUserToOU(
+	public UserDto assignUserToOU(
 			@PathVariable(value = "userId") String userId, 
-			@PathVariable(value = "ouId") String ouId) {
-		String callerId = authenticatedUserService.getCurrentUserId();
-		log.info("{} try to assign user {} to OU {}", callerId, userId, ouId);
+			@PathVariable(value = "ouId") String ouId) throws NotFoundException {
+		String currentUserId = authenticatedUserService.getCurrentUserId();
+		log.info("{} tries to assign user {} to OU {}", currentUserId, userId, ouId);
 		Optional<UserDto> optUser = userService.getUser(userId);
 		if (optUser.isEmpty()) {
-			String notFoundUser = String.format("User %s can't be assigned : user not found - %s", userId,
-					HttpStatus.NOT_FOUND.value());
-			log.warn(notFoundUser);
-			return new ResponseEntity<>(notFoundUser, HttpStatus.NOT_FOUND);
+			throw new NotFoundException("User not found");
 		}
 
-		if (!organizationUnitService.isPresent(ouId)) {
-			String notFoundOu = String.format("Organization Unit %s can't be targeted : not found - %s", ouId,
-					HttpStatus.NOT_FOUND.value());
-			log.warn(notFoundOu);
-			return new ResponseEntity<>(notFoundOu, HttpStatus.NOT_FOUND);
+		Optional<OrganizationUnitDto> ouOptional = organizationUnitService.findById(ouId);
+		if (ouOptional.isEmpty()) {
+			throw new NotFoundException("Organization unit not found");
```

f there are no side effects, couldn't User and OU services thrown NotFoundException on getEntity and let it bubble up to the controllerAdvice ?